### DISCUSSION TITLE
Moving a space to another display changes space numbering

### DISCUSSION THREAD

**Discussion by mil-ad** [asmvik/yabai#1554] [General]
Not sure if this is a bug or works-as-intented but I noticed that if I move a space to another display the space number changes and all the spaces on the origin display are shifted by one. Is there a way to keep space numbers persist so my keybindings (hyper + space_num) keep on working without me having to look up new space numbers?

**Comment by 24unix** [comment]
Did you disable rearranging?
![SCR-20221206-pv1](https://user-images.githubusercontent.com/36622631/205982760-f073975d-8fe2-4c51-9021-8cd2a9cfe2b8.png)

**Comment by mil-ad** [comment]
I have. 

On one display everything is persistent and I have space_[0-9]. When I connect an external monitor I get space_10 on that monitor which is fine. I want to be able to for instance move space 2 with a keybinding to the external monitor and have space[0-1],[3-9] on original display and space[10,2] on the external one. Is this possible?

**Reply by asmvik** [reply]
The space numbering is handled by macOS and is not something we can override. You can assign labels to spaces in yabai upon startup, which you can refer to in Yabai-related commands that operate on spaces, which will allow for the sort of keybindings that you are trying to create.

**Reply by levifig** [reply]
I wouldn't mind using labels instead of macOS' native numbering, but, for whatever reason, I couldn't get `--focus` to operate on labels… It just simply didn't do anything! Do I need to use quotes? Documentation doesn't imply I would… :\
